### ISSUE TITLE
[Bug]: terraform doesn't support valid Keyspace table complex data type

### ISSUE THREAD

**Issue by mkamel93** [hashicorp/terraform-provider-aws#35787]
### Terraform Core Version

1.7.3

### AWS Provider Version

5.35.0

### Affected Resource(s)

- AWS Keyspaces tables

### Expected Behavior

- terraform should create the table as this is a supported data type now in AWS  as shown in this screenshot 

### Actual Behavior

Trying to create an AWS keyspace table with column of type `frozen<map<text,text>>` or `set<frozen<map<text, text>>>` fails becuase the validate function is forcing the type to must consist of lower case alphanumerics and an optional list of upto two lower case alphanumerics enclosed in angle brackets `'<>'.`

### Relevant Error/Panic Output Snippet

```shell
Error: invalid value for schema_definition.0.column.1.type (The type must consist of lower case alphanumerics and an optional list of up to two lower case alphanumerics enclosed in angle brackets '<>'.)
```

### Terraform Configuration Files

https://gist.github.com/mkamel93/2979b6a23c55ce55aede496918bc1443

### Steps to Reproduce

- try to plan the attached files 
- plan will fail with the attached error

### Debug Output

this function regex needs to be updated to support that kinds of supported types

### Would you like to implement a fix?

None

**Comment by MS26** [comment]
Getting the same error message when running `validate` with the following column type `list<frozen<tuple<uuid, int>>>` is there any update on this?

**Comment by DiegoFalcon** [comment]
Getting the same error when running `terraform plan` with the column type `frozen<list<TEXT>>`. Terraform plan doesn't work when creating any nested structured type.

`schema_definition.0.column.29.type (The type must consist of lower case alphanumerics and an optional list of upto two lower case alphanumerics enclosed in angle brackets '<>'.)`

**Comment by thomty** [comment]
Experienced this same issue. Looks like an issue with the regex. It only supports up to 1 nested value, whereas keyspaces supports up to 5 by default. 

I took a crack at a modified version of the regex that will accept up to 5 layers and submitted a PR for it.

**Comment by thomty** [comment]
> Getting the same error when running `terraform plan` with the column type `frozen<list<TEXT>>`. Terraform plan doesn't work when creating any nested structured type.
> 
> `schema_definition.0.column.29.type (The type must consist of lower case alphanumerics and an optional list of upto two lower case alphanumerics enclosed in angle brackets '<>'.)`

FWIW the "frozen<list<TEXT>>" example won't work as the regex accepts only lowercase. Might be an AWS limitation, but not 100% sure.

**Comment by DiegoFalcon** [comment]
> Experienced this same issue. Looks like an issue with the regex. It only supports up to 1 nested value, whereas keyspaces supports up to 5 by default.
> 
> I took a crack at a modified version of the regex that will accept up to 5 layers and submitted a PR for it.

As you mentioned the issue is with the regex as it only supports 1 nested value. The data type `frozen<list<TEXT>>` can be easily converted to lower case, but still fails due to the nested limitations.

Could you please share the PR so we can keep track on the progress?

**Comment by thomty** [comment]
> > Experienced this same issue. Looks like an issue with the regex. It only supports up to 1 nested value, whereas keyspaces supports up to 5 by default.
> > I took a crack at a modified version of the regex that will accept up to 5 layers and submitted a PR for it.
> 
> As you mentioned the issue is with the regex as it only supports 1 nested value. The data type `frozen<list<TEXT>>` can be easily converted to lower case, but still fails due to the nested limitations.
> 
> Could you please share the PR so we can keep track on the progress?

Yep! It should be linked above as well: https://github.com/hashicorp/terraform-provider-aws/pull/38525

**Comment by wallhided** [comment]
bump

**Comment by thomty** [comment]
PR Linked above fixes the issue, but it's been almost 2 years without review from code owners. Closed PR for housekeeping purposes.

**Comment by george-zubrienko** [comment]
Really sad the fix got discontinued - will try to nudge via enterprise support, the new open-source way lol.
